src/app/timeseries-menu.component.ts

```diff
@@ -56,18 +62,20 @@ export class TimeseriesMenuComponent {
   autoScale = true;
 
   private _model: Model;
+  private dateJson: any;
 
   public GetLevels() {
     return this._model.settings.Levels;
   }
 
   public DataAvailable() {
-    return this.levelsLoaded === this.multi.length && this.levelsLoaded > 0
-      ? true
-      : false;
+    return this.levelsLoaded === this.multi.length && this.levelsLoaded > 0;
   }
 
-  public constructor(@Inject(MAT_DIALOG_DATA) public data: any) {
+  public constructor(
+    private dialogRef: MatDialogRef<TimeseriesMenuComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: any
```

Comment:
Is this how you're getting the "model" object in the "HTML" that you're passing to the below functions? It's pretty cool.